### ISSUE TITLE
Just like ClickOnce but does not work

### ISSUE THREAD

**Issue by jckpl** [Squirrel/Squirrel.Windows#445]
## I've watched this Channel 9 video 15 times

https://channel9.msdn.com/Events/dotnetConf/2015/Squirrel-for-Windows-installing-NET-apps-the-way-it-should-be
 and I've dug through 100 posts trying to get Squirrel for Windows to work. At first I was heartened when the initial install worked very well. But no matter what I do, the application will not update! 
I pride myself on being stubborn and not giving up but I've wasted days on this.
If anyone else has struggled with getting Squirrel to update your Windows application ( the whole purpose behind it BTW!) I would like to hear from you.

**Comment by willdean** [comment]
Nobody will be able to help you on the basis of this post.   You'll have to say what you've tried and what the symptoms are when it doesn't work.

**Comment by jckpl** [comment]
willdean,
Listing everything would be one long post, but here's a start:
I call this from the entry point of the app:
   static void updateApp()
        {
            Task.Run(async () =>
            {
                using (var mgr = new UpdateManager(@"\\lblstlsql1\adminsoftware$\print_barcode", "print_barcode", FrameworkVersion.Net45))
                {
                    var updateInfo = await mgr.CheckForUpdate();
                    if (updateInfo == null || updateInfo.ReleasesToApply.Count == 0)
                    {
                      
                        return;

                    }
                
                    await mgr.DownloadReleases(updateInfo.ReleasesToApply);
                    await mgr.ApplyReleases(updateInfo);


                }
            });

            
        }

**Comment by paulcbetts** [comment]
Maybe attach a debugger and see what's in `updateInfo` to start?

**Comment by jckpl** [comment]
Hi Paul,
thanks for jumping in, I really appreciate it since I was starting to have a meltdown!
I edited my replay to willdean to include two more things I tried.

To your suggestion, I'm not sure how to debug a async task. I was researching it just now. I have VS2015 and was going to try the Task Runner Window. Is that what you would do?

**Comment by paulcbetts** [comment]
I would set a breakpoint!

**Comment by jckpl** [comment]
I did. it will not hit it since (I guess) it's an async task.
Although, stepping through when I had the breakpoint set, it hit it but did not step into the task.

I was reading through this post to see what I'm doing wrong in debugging it.
http://blogs.msdn.com/b/visualstudioalm/archive/2013/07/02/debugging-asynchronous-code-in-visual-studio-2013-tasks-window.aspx

**Comment by jckpl** [comment]
something interesting did just happen, I edited my icon and when I went to AppData\Local\print_barcode 
the icon was the new version.
 
![image](https://cloud.githubusercontent.com/assets/9698102/10049368/ca3986c8-61dc-11e5-913a-3d19cb6ad772.png)

**Comment by jckpl** [comment]
with a breakpoint:
![image](https://cloud.githubusercontent.com/assets/9698102/10049488/524cf446-61dd-11e5-8126-f9415179add8.png)

after stepinto:
![image](https://cloud.githubusercontent.com/assets/9698102/10049509/6bccb4ba-61dd-11e5-952c-7a0f5c4ff6a5.png)

updateInfo never exists

**Comment by dealproc** [comment]
try setting a breakpoint on the line that says `var updateInfo = await mgr.CheckForUpdate();`... it sounds a bit asinine, but with the Task stuff, sometimes you need to.

**Comment by jckpl** [comment]
Yep, tried it. also put the Messagbox in but that doesn't work either since I guess it's not on the UI thread.

**Comment by jckpl** [comment]
I commented out the 'Task.Run(async) bit and then ran it to that point.
it hit the breakpoint and here's the value in updateInfo
+		updateInfo	Id = 1, Status = WaitingForActivation, Method = "{null}", Result = "{Not yet computed}"	System.Threading.Tasks.Task<Squirrel.UpdateInfo>
 
which is why it's should be an async task, it's apparently waiting for the update information.

**Comment by jckpl** [comment]
By the way Paul, my title for this issue was quite snarky. Sorry about that. Very rude of me.
Coding means working through many many variables and I'm sure this issue is something I'm not doing right.

**Comment by willdean** [comment]
Have you taken out an await from the CheckForUpdate call?  It looks like you're now looking at a Task<T> rather than the actual update info itself.

**Comment by jckpl** [comment]
oh yes.  since the task.run was commented out , it would not allow an 'await' for the mgr.checkforupdate()
![image](https://cloud.githubusercontent.com/assets/9698102/10051015/f33327f2-61e4-11e5-9f8d-d04e7ed30c1f.png)

**Comment by willdean** [comment]
The problem is, as you've discovered, that CheckForUpdate doesn't return the update info, it returns a Task, and you need to wait for that task to finish to get the result.

You would probably be best served by putting Squirrel to one side for a day, and really get to grips with the Task and with the async/await stuff first.

Everything by this guy http://blog.stephencleary.com/ or Stephen Toub  is worth reading.  The latter wrote this http://www.microsoft.com/en-gb/download/details.aspx?id=19957 which is worth reading 10x over.  

Don't just blunder about with the Task-based stuff because you'll get into all sorts of horrible problems.

**Comment by jckpl** [comment]
Thanks Will. good advice.
I'll keep this issue open and post what I find.

**Comment by jckpl** [comment]
Okay guys, here's something interesting that makes me think I'm missing a step in the package building process. 
I revved up my nuspec file after building my application with Release output. I changed a version number that shows up in the form text so I will know if I'm opening the new version.
I built the new package
ran releasify on the new package
started releases.
In the network share where I move all the files I deleted all the old ones. I also uninstalled the application from the target pc. I then copied my new releases files to the network share.
I double clicked on Setup.exe in the share and the old version installed. What's up with that?
